datasette/app.py

```diff
@@ -217,6 +226,27 @@ def add_database(self, name, db):
     def remove_database(self, name):
         self.databases.pop(name)
 
+    def scan_dirs(self):
+        # Recurse through self.dirs looking for new SQLite DBs
+        while True:
+            current_filepaths = {
+                d.path for d in list(self.databases.values()) if d.path is not None
+            }
+            for dir in self.dirs:
+                for filepath in Path(dir).glob("**/*.db"):
+                    if str(filepath) in current_filepaths:
+                        continue
+                    print(filepath)
+                    if is_valid_sqlite(filepath):
+                        self.add_database(
+                            str(filepath)
+                            .replace("../", "")
+                            .replace("/", "_")
+                            .replace(".db", ""),
+                            Database(self, str(filepath), is_mutable=True),
+                        )
+            time.sleep(10)
```

Comment:
This should be configurable - 10 is an OK default, but cases like https://github.com/simonw/jupyterserverproxy-datasette-demo/issues/2 where it's running in a Binder session would benefit from faster updates.